From c403e1f4f317ad8a1dd6767ba8a0fc1a6e43dd8b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Sat, 2 Nov 2024 10:21:27 +0100
Subject: [PATCH] Pass blurred image directly to texture

Instead of saving the blurred image into a temporary file and then read it
back, pass it direcly to texture via a buffer.
---
 cartridges/game_cover.py | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/cartridges/game_cover.py b/cartridges/game_cover.py
index 46dd6740c..4d73f9dd3 100644
--- a/cartridges/game_cover.py
+++ b/cartridges/game_cover.py
@@ -17,6 +17,7 @@
 #
 # SPDX-License-Identifier: GPL-3.0-or-later
 
+from io import BytesIO
 from pathlib import Path
 from typing import Optional
 
@@ -83,10 +84,11 @@ def get_blurred(self) -> Gdk.Texture:
                         .filter(ImageFilter.GaussianBlur(20))
                     )
 
-                    tmp_path = Gio.File.new_tmp(None)[0].get_path()
-                    image.save(tmp_path, "tiff", compression=None)
+                    buffer = BytesIO()
+                    image.save(buffer, "tiff", compression=None)
+                    gbytes = GLib.Bytes.new(buffer.getvalue())
 
-                    self.blurred = Gdk.Texture.new_from_filename(tmp_path)
+                    self.blurred = Gdk.Texture.new_from_bytes(gbytes)
 
                     stat = ImageStat.Stat(image.convert("L"))
 
